doc/modules/clustering.rst

```diff
@@ -1582,3 +1582,57 @@ Drawbacks
  *  Caliński, T., & Harabasz, J. (1974). "A dendrite method for cluster
     analysis". Communications in Statistics-theory and Methods 3: 1-27.
     `doi:10.1080/03610926.2011.560741 <http://dx.doi.org/10.1080/03610926.2011.560741>`_.
+
+.. _contingency_matrix:
+
+Contingency Matrix
+----------------------
+
+Contingency matrix (:func:`sklearn.metrics.cluster.contingency_matrix`)
+reports the intersection cardinality for every true/predicted cluster pair.
+The contingency matrix provides sufficient statistics for all clustering
+metrics where the samples are independent and identically distributed and
+one doesn't need to account for some instances not being clustered.
+
+Here is an example::
+
+   >>> from sklearn.metrics.cluster import contingency_matrix
+   >>> x = ["a", "a", "a", "b", "b", "b"]
+   >>> y = [0, 0, 1, 1, 2, 2]
+   >>> contingency_matrix(x, y)
+   array([[2, 1, 0],
+          [0, 1, 2]])
+
+The first row of output array indicates that there are three samples whose
+true cluster is "a", two whose predicted cluster is 0, one of whose predicted
+cluster is 1 and none of whose predicted cluster is 2. Seconds rows indicates
+that there are three samples whose true cluster is "b", none of whose
+predicted cluster is 0, one of whose predicted cluster is 1 and two of whose
+predicted cluster is 2.
+
+Confusion matrix is square matrix and the order of columns and rows is significant.
```

Comment:
"A confusion matrix for classification is a square contingency matrix where the order of rows and columns correspond to a list of classes." Please use :ref: to link to the relevant section in the user guide

Reply:
Probably addressed this okay as well.